spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -16,47 +16,28 @@
 
 package org.springframework.cloud.gcp.pubsub.integration.outbound;
 
-import com.google.cloud.pubsub.spi.v1.Publisher;
-import com.google.pubsub.v1.PubsubMessage;
-import com.google.pubsub.v1.TopicName;
 import java.io.IOException;
-import org.springframework.cloud.gcp.pubsub.integration.converters.SimpleMessageConverter;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
 import org.springframework.integration.handler.AbstractMessageHandler;
 import org.springframework.messaging.Message;
-import org.springframework.messaging.converter.MessageConversionException;
-import org.springframework.messaging.converter.MessageConverter;
 
 /**
- * Converts from internal Spring message to GCP Pub/Sub message and publishes the Pub/Sub
- * message to a Pub/Sub topic.
+ * Sends messages to Google Cloud Pub/Sub by delegating to {@link PubSubTemplate}.
  *
  * @author João André Martins
  */
 public class PubSubMessageHandler extends AbstractMessageHandler {
 
-  private Publisher publisher;
-  private MessageConverter messageConverter;
+  private PubSubTemplate pubsubTemplate;
+  private String topic;
 
-  public PubSubMessageHandler(String projectId, String topicName) throws IOException {
-    this(projectId, topicName, new SimpleMessageConverter());
-  }
-
-  public PubSubMessageHandler(String projectId, String topicName,
-      MessageConverter messageConverter) throws IOException {
-    publisher = Publisher.defaultBuilder(TopicName.create(
-        projectId, topicName)).build();
-    this.messageConverter = messageConverter;
+  public PubSubMessageHandler(String projectId, String topic) throws IOException {
+    pubsubTemplate = new PubSubTemplate(null, projectId);
+    this.topic = topic;
```

Comment:
My 2c @joaoandremartins is to not confuse the PubSubTemplate with the Handler. What @artembilan is describing is a a very known pattern in SI. Topics would be in the header, now, the Handler has the decision what to do with that topic name, and in our case it would just pass it to the `send(topic, PubSubMessage)` inside the PubsubTemplate, right?

The internals of the PubSubTemplate will need to handle how create publishers and cache them, but this is a different story.